### PR TITLE
[extractor/indavideo] Fix video extraction

### LINKED ISSUE #7190: indavideo.hu is broken
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

Downloading this video: https://indavideo.hu/video/A_regi_Toyota_mint_nehezfuvarozo

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://indavideo.hu/video/A_regi_Toyota_mint_nehezfuvarozo
[debug] Command-line config: ['-vU', 'https://indavideo.hu/video/A_regi_Toyota_mint_nehezfuvarozo']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.11.2 (CPython x86_64 64bit) - macOS-10.11.6-x86_64-i386-64bit (OpenSSL 1.1.1s  1 Nov 2022)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.17, brotli-1.0.9, certifi-2022.12.07, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[generic] Extracting URL: https://indavideo.hu/video/A_regi_Toyota_mint_nehezfuvarozo
[generic] A_regi_Toyota_mint_nehezfuvarozo: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] A_regi_Toyota_mint_nehezfuvarozo: Extracting information
[debug] Looking for embeds
[debug] Identified a IndavideoEmbed embed
[IndavideoEmbed] Extracting URL: https://embed.indavideo.hu/player/video/06fc090308
[IndavideoEmbed] 06fc090308: Downloading JSON metadata
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 3626240: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://index1-int.indavideo.hu/indavideo/003/626/240/62ecf025910c45.04194424.360.mp4?channel=main&zone=int&token=1873d4344ecd33e851ee31c7789af838"
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 3247, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 2970, in dl
    return fd.download(name, new_info, subtitle)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/downloader/common.py", line 444, in download
    ret = self.real_download(filename, info_dict)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/downloader/http.py", line 376, in real_download
    establish_connection()
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/downloader/http.py", line 129, in establish_connection
    ctx.data = self.ydl.urlopen(request)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 3742, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 525, in open
    response = meth(req, response)
               ^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 634, in http_response
    response = self.parent.error(
               ^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 563, in error
    return self._call_chain(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 496, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

### PR DESCRIPTION
**Pull request by aky-01**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

- Fix video extraction
- Remove dead links in example urls section
- Some code cleanup

Fixes #7190

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🛠️🔒🧹

Fix indavideo extraction by adding timestamp, using HTTPS, and simplifying code. Update example and test URLs in `indavideo.py`.

> _Oh we're the crew of the indavideo.py_
> _And we've fixed the extraction of the clips_
> _We've added a `timestamp` and used HTTPS_
> _And we've simplified some of the scripts_

### Walkthrough
* Import `time_seconds` function from `utils` module to append timestamp to video data URL ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaR8))
* Use HTTPS instead of HTTP for example URLs, `_EMBED_REGEX`, and test URLs to avoid redirection or mixed content issues ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL14-R22), [link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL39-R42))
* Add timestamp parameter to video data URL using `time_seconds` function to bypass caching issue ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL50-R52))
* Remove redundant assignment of `title` from video data and retrieve it from the video itself ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL50-R52), [link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL106-R100))
* Remove unnecessary check and append of `video_file` to `video_urls` list and avoid duplicates or invalid URLs ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL63-L65))
* Simplify logic of extracting `height` and `token` from `filesh` dictionary and remove redundant check of `filesh` being truthy ([link](https://github.com/yt-dlp/yt-dlp/pull/8129/files?diff=unified&w=0#diff-801de90d10401cb3ee744f95bbd39dab532e58d698a3542284458fb12b1e3dfaL81-R81))

</details>

### CODE REVIEW

File: yt_dlp/extractor/indavideo.py [hunk 1]
```diff
@@ -78,13 +73,12 @@ def _real_extract(self, url):
...
-            if filesh:
-                if not height:
-                    continue
-                token = filesh.get(compat_str(height))
-                if token is None:
-                    continue
-                video_url = update_url_query(video_url, {'token': token})
+            if not height and len(filesh) == 1:
+                height = int(list(filesh.keys())[0])
+            token = filesh.get(compat_str(height))
```
Comment (coletdjnz) [inline yt_dlp/extractor/indavideo.py:78]:
```suggestion
            token = filesh.get(str(height))
```

can remove the compat_str now

File: yt_dlp/extractor/indavideo.py [hunk 2]
```diff
@@ -78,13 +73,12 @@ def _real_extract(self, url):
...
                 r'\.(\d{3,4})\.mp4(?:\?|$)', video_url, 'height', default=None))
-            if filesh:
-                if not height:
-                    continue
-                token = filesh.get(compat_str(height))
-                if token is None:
-                    continue
-                video_url = update_url_query(video_url, {'token': token})
+            if not height and len(filesh) == 1:
+                height = int(list(filesh.keys())[0])
```
Comment (coletdjnz) [inline yt_dlp/extractor/indavideo.py:77]:
```suggestion
                height = int_or_none(list(filesh.keys())[0])
```

File: yt_dlp/extractor/indavideo.py [hunk 3]
```diff
@@ -5,21 +5,21 @@
     parse_age_limit,
     parse_iso8601,
     update_url_query,
+    time_seconds,
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:8]:
```suggestion
    time_seconds,
    update_url_query,
```

File: yt_dlp/extractor/indavideo.py [hunk 4]
```diff
@@ -5,21 +5,21 @@
...
-    #   http://auto.indavideo.hu/video/Sajat_utanfutoban_a_kis_tacsko
-    #   http://erotika.indavideo.hu/video/Amator_tini_punci
-    #   http://film.indavideo.hu/video/f_hrom_nagymamm_volt
-    #   http://palyazat.indavideo.hu/video/Embertelen_dal_Dodgem_egyuttes
-    _EMBED_REGEX = [r'<iframe[^>]+\bsrc=["\'](?P<url>(?:https?:)?//embed\.indavideo\.hu/player/video/[\da-f]+)']
+    #   https://indavideo.hu/video/Vicces_cica_1
+    #   https://index.indavideo.hu/video/Hod_Nemetorszagban
+    #   https://auto.indavideo.hu/video/Sajat_utanfutoban_a_kis_tacsko
+    #   https://film.indavideo.hu/video/f_farkaslesen
+    #   https://palyazat.indavideo.hu/video/Embertelen_dal_Dodgem_egyuttes
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:19]: could you add one of these as a webpage test?

e.g.
```py
    _WEBPAGE_TESTS = [{
        'url': 'https://indavideo.hu/video/Vicces_cica_1',
        'info_dict': {
            'id': 'id_goes_here',
            'ext': 'mp4',
            # etc etc
        },
    }]
```

File: yt_dlp/extractor/indavideo.py [hunk 5]
```diff
@@ -36,22 +36,20 @@ class IndavideoEmbedIE(InfoExtractor):
...
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
         video = self._download_json(
-            'https://amfphp.indavideo.hu/SYm0json.php/player.playerHandler.getVideoData/%s' % video_id,
+            'https://amfphp.indavideo.hu/SYm0json.php/player.playerHandler.getVideoData/%s/?_=%d' % (video_id, time_seconds()),
             video_id)['data']
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:51]:
```suggestion
            f'https://amfphp.indavideo.hu/SYm0json.php/player.playerHandler.getVideoData/{video_id}/',
            video_id, query={'_': time_seconds()})['data']
```

File: yt_dlp/extractor/indavideo.py [hunk 6]
```diff
@@ -78,13 +73,12 @@ def _real_extract(self, url):
         for video_url in video_urls:
             height = int_or_none(self._search_regex(
                 r'\.(\d{3,4})\.mp4(?:\?|$)', video_url, 'height', default=None))
-            if filesh:
-                if not height:
-                    continue
-                token = filesh.get(compat_str(height))
-                if token is None:
-                    continue
-                video_url = update_url_query(video_url, {'token': token})
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:87]: The old code implies that the `token` is optional and `video['filesh']` may be non-existent, and as such the old token code was completely non-fatal. Do you know for certain that every video requires a token now?

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/indavideo.py [hunk 7]
```diff
@@ -78,13 +73,12 @@ def _real_extract(self, url):
...
+            if not height and len(filesh) == 1:
+                height = int(list(filesh.keys())[0])
+            token = filesh.get(compat_str(height))
+            if token is None:
+                continue
+            video_url = update_url_query(video_url, {'token': token})
             formats.append({
                 'url': video_url,
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:83]:
```suggestion
            formats.append({
                'url': update_url_query(video_url, {'token': token}),
```

File: yt_dlp/extractor/indavideo.py (hunk 6 above)
Comment (aky-01) [reply yt_dlp/extractor/indavideo.py:87]: > The old code implies that the `token` is optional and `video['filesh']` may be non-existent, and as such the old token code was completely non-fatal. Do you know for certain that every video requires a token now?

My experience is that all video links requires token for a long time.

File: yt_dlp/extractor/indavideo.py [hunk 8]
```diff
@@ -36,21 +35,36 @@ class IndavideoEmbedIE(InfoExtractor):
...
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'uploader': 'Jet_Pack',
+            'uploader_id': '491217',
+            'timestamp': 1390821212,
+            'upload_date': '20140127',
+            'duration': 7,
+            'age_limit': 0,
+            'tags': ['cica', 'Jet_Pack'],
+        },
+    }, {
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:54]: This is not what https://github.com/yt-dlp/yt-dlp/pull/8129/files#r1328046327 was asking for

Add a new class var, `_WEBPAGE_TESTS`, and add this test to it with its original URL (https://indavideo.hu/video/Vicces_cica_1)

We need to test embed extraction

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/indavideo.py (hunk 8 above)
Comment (aky-01) [reply yt_dlp/extractor/indavideo.py]: Sorry I missed it. Fixed in [this](https://github.com/yt-dlp/yt-dlp/pull/8129/commits/d57554f1acd6812f8878c981089106381476ddd7) commit.

File: yt_dlp/extractor/indavideo.py [hunk 9]
```diff
@@ -60,9 +74,6 @@ def _real_extract(self, url):
         elif isinstance(video_files, dict):
             video_urls.extend(video_files.values())
 
-        video_file = video.get('video_file')
-        if video:
-            video_urls.append(video_file)
```
Comment (bashonly) [inline yt_dlp/extractor/indavideo.py:65]: pretty sure it should've been `if video_file:`, but are we sure we're not losing any formats by removing this code?

**Review by bashonly — APPROVED** [review]

**Pull request merged**